crates/deltalake-core/src/writer/json.rs

```diff
@@ -46,7 +45,7 @@ pub(crate) struct DataArrowWriter {
     writer_properties: WriterProperties,
     buffer: ShareableBuffer,
     arrow_writer: ArrowWriter<ShareableBuffer>,
-    partition_values: HashMap<String, Option<String>>,
+    partition_values: BTreeMap<String, Scalar>,
```

Comment:
Why the change from a HashMap to BTree?

Reply:
For the partition values, ordering matters in several places. Previously we were using a combination of the 'partition_values' on metadata and the partition values HashMap. With this change we can rely on the insertion order and no longer need to track the values vector.